api/tests/unit/domain/services/recommendation-service_test.js

```diff
@@ -0,0 +1,112 @@
+        const maxSkillLevelInTargetProfile = 6;
+        const skills = [
+          { id: 1, difficulty: 1 },
+          { id: 2, difficulty: 2 },
+          { id: 3, difficulty: 3 },
+          { id: 4, difficulty: 6 },
+        ];
+
+        // when
+        const score = recommendationService.computeRecommendationScore(skills, maxSkillLevelInTargetProfile, []);
+
+        // then
+        expect(score).to.equal(30);
+      });
+    });
+
+    context('when all skills have been validated', () => {
+      it('should return 100 when the tube has the maximum level', () => {
+        // given
+        const maxSkillLevelInTargetProfile = 3;
+        const skills = [
+          { id: 1, difficulty: 1 },
+          { id: 2, difficulty: 2 },
+          { id: 3, difficulty: 3 },
+        ];
+
+        const knowledgeElements = [
+          { skillId: 1 },
+          { skillId: 2 },
+          { skillId: 3 },
+        ];
+        // when
+        const score = recommendationService.computeRecommendationScore(skills, maxSkillLevelInTargetProfile, knowledgeElements);
+
+        // then
+        expect(score).to.equal(100);
+      });
+
+      it('should return 90 when the tube does not have the maximum level', () => {
+        // given
+        const maxSkillLevelInTargetProfile = 6;
+        const skills = [
+          { id: 1, difficulty: 1 },
+          { id: 2, difficulty: 4 }
+        ];
+
+        const knowledgeElements = [
+          { skillId: 1 },
+          { skillId: 2 }
+        ];
+        // when
+        const score = recommendationService.computeRecommendationScore(skills, maxSkillLevelInTargetProfile, knowledgeElements);
+
+        // then
+        expect(score).to.equal(90);
+      });
+    });
+
+    context('when only some skills have been validated', () => {
+      it('should return 70 ', () => {
+        // given
+        const maxSkillLevelInTargetProfile = 6;
+        const skills = [
+          { id: 1, difficulty: 1 },
+          { id: 2, difficulty: 2 },
+          { id: 3, difficulty: 3 },
+          { id: 4, difficulty: 5 },
+        ];
+
+        const knowledgeElements = [
+          { skillId: 1 },
+          { skillId: 2 },
+          { skillId: 3 },
+        ];
+        // when
+        const score = recommendationService.computeRecommendationScore(skills, maxSkillLevelInTargetProfile, knowledgeElements);
+
+        // then
+        expect(score).to.equal(70);
+      });
+
+      it('should return 48.3', () => {
+        // given
+        const maxSkillLevelInTargetProfile = 6;
+        const skills = [
+          { id: 1, difficulty: 1 },
+          { id: 2, difficulty: 2 },
+          { id: 3, difficulty: 6 }
+        ];
+
+        const knowledgeElements = [
+          { skillId: 1 }
+        ];
+        // when
+        const score = recommendationService.computeRecommendationScore(skills, maxSkillLevelInTargetProfile, knowledgeElements);
+
+        // then
+        expect(score).to.be.within(48.33, 48.34);
```

Comment:
Chercher comment tester sur une précision

Reply:
`.should.be.closeTo(6.666, 0.001);`

Reply:
done